tests/unit/dependencies.js

```diff
@@ -0,0 +1,167 @@
+                    return expectModuleToHaveInjectable(moduleName, depName, 'service');
+                },
+            };
+        },
+        toHaveInjectableController: function () {
+            return {
+                compare: function (moduleName, depName) {
+                    return expectModuleToHaveInjectable(moduleName, depName, 'controller');
+                },
+            };
+        },
+        toHaveInjectableDirective: function () {
+            return {
+                compare: function (moduleName, depName) {
+                    return expectModuleToHaveInjectable(moduleName, depName, 'directive');
+                },
+            };
+        },
+    };
+
+    describe('Dependencies', function () {
+
+        beforeEach(function () {
+            jasmine.addMatchers(matchers);
+        });
+
+        // Test the matchers for expected failure.
+        describe('expected failure', function () {
+
+            beforeEach(function () {
+                angular.module('missing-deps', [])
+                    .factory('factory', function (missingInjectable) {
+                        return missingInjectable;
+                    })
+                    .service('service', function (missingInjectable) {
+                        return missingInjectable;
+                    })
+                    .controller('controller', function (missingInjectable) {
+                        this.foo = missingInjectable;
+                    })
+                    .directive('directive', function (missingInjectable) {
+                        return {
+                            link: function (scope) {
+                                scope.foo = missingInjectable;
+                            },
+                        };
+                    });
+                module('missing-deps');
+            });
+
+            describe('for module missing-deps', function () {
+                it('should error when injected', function () {
+                    expect('missing-deps').not.toHaveInjectableFactory('factory');
+                    expect('missing-deps').not.toHaveInjectableService('service');
+                    expect('missing-deps').not.toHaveInjectableController('controller');
+                    expect('missing-deps').not.toHaveInjectableDirective('directive');
+                });
+            });
+
+        });
+
+        // Load each module and inject all registered services/factories to
+        // uncover those using dependencies they haven't listed in the module.
+        angular.registeredModules.forEach(function (moduleName) {
+            describe('for module ' + moduleName, function () {
+
+                beforeEach(function () {
+                    module(moduleName);
+                });
+
+                it('should not error when injected', function () {
+                    var moduleDefinition = angular.module(moduleName);
+                    moduleDefinition._invokeQueue.forEach(function (providerArgs) {
```

Comment:
It is doing a lot, yes. There's an individual suite for every module: the code here is creating multiple tests from one loop function, which allows us not only to be DRY, but also to ensure no module is missed. A static list of tests would mean any new module would not be automatically included.

I like the automatic inclusion, and it helps the meaning of this test: to let the dev know when module dependencies haven't been listed, so the `src` changes in d82266f2ff61b80f0458257b1f2815ed0fbf42c0 aren't needed down the line, and tests don't fail confusingly (the error output for missing module dependencies is often truncated so much as to be useless).